tests/unit/stress/test_preprocesses.py

```diff
@@ -0,0 +1,54 @@
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
+import pandas as pd
+import pytest
+
+from tests.stress.buildpreprocess import generate_maximally_complex_csv
+
+
+def test_generated_csv_exists():
+    """
+        This function checks that `_generate_maximally_complex_csv` generates a csv file and check that the files exist at the desired location
+    """
+    with TemporaryDirectory() as tmp:
+
+        tmp_dir = Path(tmp)
+
+        numb_rows = 100
+
+        generate_maximally_complex_csv(numb_rows, tmp_dir)
+
+
+        assert (tmp_dir / "stress.csv").exists()
+
+
+def test_generated_csv_shape_and_num_unique_values():
```

Comment:
Why not use `assert_frame_equal` provided by pandas test utils to assert the entire data frame?

Reply:
Reworked test so that it uses `assert_frame_equals`. I have kept in some of the original assertions because I think it will be useful to see if these specific checks fail due to future changes.